src/services/azure/signer.rs

```diff
@@ -0,0 +1,291 @@
+        self
+    }
+
+    pub fn access_name(&mut self, access_name: &str) -> &mut Self {
+        self.credential.set_access_name(access_name);
+        self
+    }
+
+    pub async fn build(&mut self) -> Result<Signer> {
+        let credential = if self.credential.is_valid() {
+            Some(self.credential.clone())
+        } else {
+            // Make sure credential load chain has been set before checking.
+            if self.credential_load.is_empty() {
+                self.credential_load.push(EnvLoader::default());
+            }
+
+            self.credential_load.load_credential().await?
+        };
+        debug!("credential has been set to: {:?}", &credential);
+        Ok(Signer {
+            credential: Arc::new(RwLock::new(credential)),
+            credential_load: mem::take(&mut self.credential_load),
+        })
+    }
+}
+#[derive(Default)]
+pub struct Signer {
+    credential: Arc<RwLock<Option<Credential>>>,
+    credential_load: CredentialLoadChain,
+}
+
+//since access_account and shared_key should be kept secret,so debug trait is unimplemented
+impl Debug for Signer {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Signer {{ unimplement!}}")
+    }
+}
+
+impl Signer {
+    pub fn builder() -> Builder {
+        Builder::default()
+    }
+
+    /// Load credential via credential load chain specified while building.
+    async fn credential(&self) -> Result<Option<Credential>> {
+        // Return cached credential if it's valid.
+        match self.credential.read().await.clone() {
+            None => return Ok(None),
+            Some(cred) => {
+                if cred.is_valid() {
+                    return Ok(Some(cred));
+                }
+            }
+        }
+
+        if let Some(cred) = self.credential_load.load_credential().await? {
+            let mut lock = self.credential.write().await;
+            *lock = Some(cred.clone());
+            Ok(Some(cred))
+        } else {
+            // We used to get credential correctly, but now we can't.
+            // Something must happened in the running environment.
+            Err(anyhow!("credential should be loaded but not"))
+        }
+    }
+
+    pub async fn sign(&self, request: &mut impl SignableRequest) -> Result<()> {
+        let host = request.host();
+        let path = request.path();
+        let option_query = request.query();
+        let url:Url = match option_query{
+            Some(query) =>Url::parse(&format!("https://{}{}?{}", host, path,query)).expect("parsing url success"),
+            _ => Url::parse(&format!("https://{}{}", host, path)).expect("parsing url success")
+        };
+
+        let method = request.method().clone();
+
+        // account = credential.access_name, keep variable nameaccount aligning with azure sdk for rust
+        // refer https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/storage/src/core/clients/storage_account_client.rs
```

Comment:
Please refer to azure storage's docs directly.

Reply:
Ok

Reply:
Which one do you mean? [Microsoft docs](https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key) use Shared key  but [ go sdk](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/storage/azblob)  use account_key ，I lean towards the latter, account_name and account_key

Reply:
LGTM, let's use `account_name` and `account_key`.